### PR TITLE
RATIS-1461. Add ratis-shell documents

### PR DESCRIPTION
**Pull request by codings-dan**
see https://issues.apache.org/jira/browse/RATIS-1461

### CODE REVIEW

**Comment by codings-dan** [comment]
@szetszwo Can you help review this patch, thanks!

File: ratis-docs/content/cli.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+
+# Ratis-shell 
+Ratis's commad line interface provides operations to manage the ratis server. You can invoke the following command line 
+utility to get all the subcommands:
+
+```shell
+$ cd $PATH_TO_RATIS/ratis-assembly/target/apache-ratis-$version-SNAPSHOT
```
Comment (szetszwo) [inline ratis-docs/content/cli.md:28]: Let's remove this line.  If we want to talk about the path and other setup such as classpath, we should have a separated section to include more details.

File: ratis-docs/content/cli.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+
+# Ratis-shell 
+Ratis's commad line interface provides operations to manage the ratis server. You can invoke the following command line 
+utility to get all the subcommands:
```
Comment (szetszwo) [inline ratis-docs/content/cli.md:25]: Let's change this to
```
Ratis-shell is the command line interface of Ratis.
The following command can get all the subcommands:
```

File: ratis-docs/content/cli.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+```shell
+$ cd $PATH_TO_RATIS/ratis-assembly/target/apache-ratis-$version-SNAPSHOT
+$ ./bin/ratis sh
```
Comment (szetszwo) [inline ratis-docs/content/cli.md:29]: Let's simply use
```
$ ratis sh
```
We may assume `ratis` is already in the PATH.

File: ratis-docs/content/cli.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+## Operations:
+
+### election
+
+The `election` command manage ratis leader election, it has four subcommands：`transfer`、`stepDown`、`pause`、`resume`
+
+```
+# Switch the ratis leader to the specified ratis server(its address is <HOSTNAME:PORT>).
+$ ./bin/ratis sh election transfer -peers <PEER0_HOST:PEER0_PORT,PEER1_HOST:PEER1_PORT,PEER2_HOST:PEER2_PORT> -address <HOSTNAME:PORT> [-groupid <RAFT_GROUP_ID>]
```
Comment (szetszwo) [inline ratis-docs/content/cli.md:45]: Let's use a shorter peer list `<P0_HOST:P0_PORT,P1_HOST:P1_PORT,P2_HOST:P2_PORT>`.

File: ratis-docs/content/cli.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+Usage: ratis sh [generic options]
+         [election [transfer] [stepDown] [pause] [resume]]         
+         [group [info] [list]]                                     
+         [peer [add] [remove] [setPriority]]                       
+         [snapshot [create]]  
+```
+
+## Operations:
+
+### election
```
Comment (szetszwo) [inline ratis-docs/content/cli.md:39]: ## election
The `election` command manages leader election. 
It has the following subcommands:
`transfer`, `stepDown`, `pause`, `resume`

### election transfer
Transfer a group leader to the specified server.
```
$ ratis sh election transfer -peers <P0_HOST:P0_PORT,P1_HOST:P1_PORT,P2_HOST:P2_PORT> -address <HOSTNAME:PORT> [-groupid <RAFT_GROUP_ID>]
```

### election stepDown
Make a group leader of the given group step down its leadership.
```
$ ratis sh election stepDown -peers <P0_HOST:P0_PORT,P1_HOST:P1_PORT,P2_HOST:P2_PORT> [-groupid <RAFT_GROUP_ID>]
```

### election pause
Pause leader election at the specified server.
Then, the specified server would not start a leader election.
```
$ ratis sh election pause -peers <P0_HOST:P0_PORT,P1_HOST:P1_PORT,P2_HOST:P2_PORT> -address <HOSTNAME:PORT> [-groupid <RAFT_GROUP_ID>]
```

### election resume
Resume leader election at the specified server.
```
$ ratis sh election resume -peers <P0_HOST:P0_PORT,P1_HOST:P1_PORT,P2_HOST:P2_PORT> -address <HOSTNAME:PORT> [-groupid <RAFT_GROUP_ID>]
```

File: ratis-docs/content/cli.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
+---
+title: Ratis-shell
+menu: main
+weight: -10
+---
```
Comment (szetszwo) [inline ratis-docs/content/cli.md:5]: These few lines seem not working as expected.

**Review by szetszwo — COMMENTED** [review]
@codings-dan , thanks a lot for working on this.

Let's add the election section first. See https://issues.apache.org/jira/secure/attachment/13043081/637_review.patch

File: ratis-docs/content/cli.md (hunk 1 above)
Comment (codings-dan) [reply ratis-docs/content/cli.md]: done, I will add additional documentation.

File: ratis-docs/content/cli.md (hunk 2 above)
Comment (codings-dan) [reply ratis-docs/content/cli.md]: Thanks for helping change the description

File: ratis-docs/content/cli.md (hunk 3 above)
Comment (codings-dan) [reply ratis-docs/content/cli.md]: done

File: ratis-docs/content/cli.md (hunk 4 above)
Comment (codings-dan) [reply ratis-docs/content/cli.md]: done

**Comment by codings-dan** [comment]
@szetszwo Thanks for helping review the patch, PTAL agian, thank you!

File: ratis-docs/content/cli.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+```shell
+$ ratis sh
+Usage: ratis sh [generic options]
+         [election [transfer] [stepDown] [pause] [resume]]
+         [group [info] [list]]
+         [peer [add] [remove] [setPriority]]
+         [snapshot [create]]
+```
+Note: Ratis-shell is currently only experimental, and compatibility story is not considered for the time being
```
Comment (szetszwo) [inline ratis-docs/content/cli.md:30]: @codings-dan , thanks for the update.  Let's move the Note up and highlight it.
```
# Ratis-shell

Ratis-shell is the command line interface of Ratis.

> **Note**:
> Ratis-shell is currently only **experimental**.
> The compatibility story is not considered for the time being.

The following command can be invoked in order to get the basic usage:

```shell
$ ratis sh
Usage: ratis sh [generic options]
         [election [transfer] [stepDown] [pause] [resume]]
         [group [info] [list]]
         [peer [add] [remove] [setPriority]]
         [snapshot [create]]
```

Comment (codings-dan) [reply ratis-docs/content/cli.md]: done

**Comment by codings-dan** [comment]
@szetszwo PTAL, thank you!

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.

**Comment by szetszwo** [comment]
Since this is a pure doc change, I will merge this without waiting the tests.

**Pull request merged**